rfd/XXXX-scopes.md

```diff
@@ -0,0 +1,570 @@
+resource itself or a descendant scope.  For example, a `scoped_role` defined at `/staging` could be assigned to users
+at `/staging` or `/staging/west`, but not at `/prod` or `/prod/west`. This ensures that role editing privileges in
+one scope cannot be used to affect permissions in another scope.
+
+As an additional layer of control, it will be possible to constrain the assignable scopes of a scoped role to
+an explicitly defined subset. For example, a scoped role defined at `/staging` could be
+configured to be assignable at `/staging/west` and `/staging/east`, but not at `/staging` or `/staging/central`
+by specifying `assignable_scopes: ["/staging/west", "/staging/east"]` in the role spec. This has a few important
+benefits. First, by allowing this extra layer of control, we allows scoped roles to be used in a manner similar
+to a "role template". If two different scopes need similar roles, a parent scope can provide a common definition.
+Second, separating the concerns of scoping role definition and assignment ensures that admins can accurately express
+intent and prevent misuse.
+
+A `scoped_role` resource for scoped admins might look like something like this:
+
+```yaml
+kind: scoped_role
+metadata:
+  name: staging-admin
+  description: Basic administrative privileges for staging env admins
+scope: /staging
+spec:
+  assignable_scopes:
+    - /staging/west
+    - /staging/east
+  allow:
+    rules:
+    - kind: scoped_token
+      verbs: [create, read, update, delete]
+    - kind: scoped_role
+      verbs: [create, read, update, delete]
+    - kind: scoped_role_assignment
+      verbs: [create, read, update, delete]
+version: v1
+```
+
+Note that other than the `scope` and `spec.assignable_scopes` fields, this is identical to a standard teleport role
+definition.  Keeping this parity ensures ease of transition and reduces cognitive load for users familiar with
+existing teleport concepts. It also saves us considerable effort since most of the existing role evaluation logic
+can be reused.
+
+
+### Scoped Tokens and Joining
+
+The scope of an agent will be determined by their join tokens. Scoped administrators will be able to create join tokens
+assigned to scopes where they have `token:create` permissions. When an agent joins the teleport cluster using a scoped
+join token, the agent will be automatically assigned that scope.
+
+Creation of scoped join tokens will mirror the existing token creation API where possible. Ex:
+
+```shell
+$ tctl scoped token add --type=node --scope=/staging/west
+```
+
+As an additional nicety, we will allow scoped admins to omit the `--scope` parameter and have it default to their currently
+pinned scope. This will help admins working with more flat (namespace-like) heirarchies to avoid needing to reason about
+scoping too much.
+
+The resulting token value will be usable in the same way an ordinary join token works today. Ex:
+
+```shell
+$ teleport start --token=some-token-value ...
```

Comment:
> we also need to ensure that agents themselves cannot be used to "reach up" and affect access outside
of their scope

Does the scoped token issue  `tctl scoped token add --type=node --scope=/staging/west` 


Does a user withy any scope with raw full buildIN teleport `access` role will still be able to connect to this agent. 

If yes the `--scope=/staging/west` doesn't restrict the permission for a identity to obtain not scoped resources like raw roles. 

The token with the `--scope=/staging/west` can still obtain a cert ssh cert with unscoped access.